src/deadline_test_fixtures/deadline/worker.py

```diff
@@ -674,32 +657,64 @@ def configure_agent_user_environment(
                 f"runuser -l {config.agent_user} -s /bin/bash -c 'aws configure add-model --service-model file://{config.service_model_path}'"
             )
 
-        if config.no_local_session_logs:
-            cmds.append(
-                f"runuser -l {config.agent_user} -s /bin/bash -c 'echo export DEADLINE_WORKER_LOCAL_SESSION_LOGS=false >> ~/.bashrc'",
-            )
-        if os.environ.get("DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE"):
-            LOG.info(
-                f"Using DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE: {os.environ.get('DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE')}"
-            )
-            cmds.append(
-                f"runuser -l {config.agent_user} -s /bin/bash -c 'echo export DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE={os.environ.get('DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE')} >> ~/.bashrc'",
-            )
+        allow_instance_profile = os.environ.get("DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE", None)
+        endpoint_url_deadline = os.environ.get("AWS_ENDPOINT_URL_DEADLINE", None)
 
-        if os.environ.get("AWS_ENDPOINT_URL_DEADLINE"):
-            LOG.info(
-                f"Using AWS_ENDPOINT_URL_DEADLINE: {os.environ.get('AWS_ENDPOINT_URL_DEADLINE')}"
+        if (
+            config.no_local_session_logs
+            or allow_instance_profile is not None
+            or endpoint_url_deadline is not None
+        ):
+            # Create a systemd override config file
+            cmds.extend(
+                [
+                    "mkdir -p /etc/systemd/system/deadline-worker.service.d/",
+                    'echo "[Service]" > /etc/systemd/system/deadline-worker.service.d/config.conf',
+                ]
             )
-            cmds.append(
-                f"runuser -l {config.agent_user} -s /bin/bash -c 'echo export AWS_ENDPOINT_URL_DEADLINE={os.environ.get('AWS_ENDPOINT_URL_DEADLINE')} >> ~/.bashrc'",
+
+            # Add environment directives to apply the configuration
+            cmds.extend(
+                [
+                    f'echo "Environment=AWS_REGION={config.region}" >> /etc/systemd/system/deadline-worker.service.d/config.conf',
+                    f'echo "Environment=AWS_DEFAULT_REGION={config.region}" >> /etc/systemd/system/deadline-worker.service.d/config.conf',
+                ]
             )
```

Comment:
After looking at this again, I think we want this applied unilaterally, so I need to restructure the branching here.

I will put up another revision after testing.

Reply:
Applied this fix